javaslang/src/test/java/javaslang/control/OptionTest.java

```diff
@@ -190,6 +190,30 @@ public void shouldReturnAlternativeOnOrElseSupplierIfValueIsNotDefined() {
         assertThat(Option.none().orElse(() -> opt)).isSameAs(opt);
     }
 
+    @Test
+    public void shouldReturnSelfOnOrElseOfIfValueIsPresent() {
```

Comment:
From what I understood, this is meant to use a provider without an `Option` wrapping?
(using `import static javaslang.API.*;`)
```java
Option(value).orElseOf(0);
```
is the same as
```
Option(value).orElse(Some(0));
```
, right?

@danieldietrich, do you think this simplifies things?